pkg/fanal/analyzer/language/java/sbt/lockfile.go

```diff
@@ -0,0 +1,86 @@
+package sbt
+
+import (
+	"context"
+	"io"
+	"io/fs"
+	"os"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/dependency/parser/sbt/lockfile"
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer/language"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/utils/fsutils"
+)
+
+func init() {
+	analyzer.RegisterPostAnalyzer(analyzer.TypeSbtLock, newSbtDependencyLockAnalyzer)
+}
+
+const (
+	version        = 1
+	fileNameSuffix = "sbt.lock"
+)
+
+// sbtDependencyLockAnalyzer analyzes '*.sbt.lock'
+type sbtDependencyLockAnalyzer struct {
+	parser language.Parser
+}
+
+func newSbtDependencyLockAnalyzer(_ analyzer.AnalyzerOptions) (analyzer.PostAnalyzer, error) {
+	return &sbtDependencyLockAnalyzer{
+		parser: lockfile.NewParser(),
+	}, nil
+}
+
+func (a sbtDependencyLockAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysisInput) (*analyzer.AnalysisResult, error) {
+	required := func(path string, d fs.DirEntry) bool {
+		return a.Required(path, nil)
+	}
+
+	var apps []types.Application
+	var err error
+	err = fsutils.WalkDir(input.FS, ".", required, func(filePath string, _ fs.DirEntry, r io.Reader) error {
+		var app *types.Application
+		app, err = language.Parse(types.Sbt, filePath, r, a.parser)
+		if err != nil {
+			return xerrors.Errorf("%s parse error: %w", filePath, err)
+		}
+
+		if app == nil {
+			// no dependencies - add empty application
+			app = &types.Application{
+				Type:     types.Sbt,
+				FilePath: filePath,
+				Packages: types.Packages{},
+			}
+		}
```

Comment:
We don't add empty apps.
e.g.:
https://github.com/aquasecurity/trivy/blob/6e7f62d2debbb4acbeb7ad58ecd8ac7379f19e40/pkg/fanal/analyzer/language/java/gradle/lockfile.go#L63-L65